# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/xul/XULPopupElement.cpp
# Commit: 983ae0fe504a
# Full Hash: 983ae0fe504a0094040eb66c48d49298d977d6be
# Author: stransky <stransky@redhat.com>
# Date: 2023-02-20 16:51:21
# Description:
#   Bug 1817419 [Linux] Make XULPopupElement::IsWaylandDragSource() more robust r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D170202
# ==============================================================================

diff -r 0390c6488ad2 -r 983ae0fe504a dom/xul/XULPopupElement.cpp
--- a/dom/xul/XULPopupElement.cpp	Mon Feb 20 10:05:51 2023 +0000
+++ b/dom/xul/XULPopupElement.cpp	Mon Feb 20 11:09:09 2023 +0000
@@ -324,8 +324,8 @@
 
 bool XULPopupElement::IsWaylandDragSource() const {
 #ifdef MOZ_WAYLAND
-  nsMenuPopupFrame* menuPopupFrame = do_QueryFrame(GetPrimaryFrame());
-  return menuPopupFrame->IsDragSource();
+  nsMenuPopupFrame* f = do_QueryFrame(GetPrimaryFrame());
+  return f && f->IsDragSource();
 #else
   return false;
 #endif
